src/pudl/transform/ferc1.py

```diff
@@ -6509,3 +6540,135 @@ def plants_steam_ferc1(
+    differ based on the ``xbrl_factoid``. So we first rename all of the columns which
+    contain the ``xbrl_factoid`` using :func:`table_to_xbrl_factoid_name` rename
+    dictionary.
+
+    Note: It is theoretically possible that we could compile these dimensions based on
+    strictly the parameters (such as using info from ``params.categorize_strings``).
+    Because we did not explictly restrict ourselves to only add categories that are
+    definitely found in the data, we went with using the data tables instead. Also since
+    we added the ``xbrl_factoid`` into the dimensions
+    """
+    table_to_xbrl_factoid_name_dict = table_to_xbrl_factoid_name()
+    tbls = {
+        name: df.assign(table_name=name).rename(
+            columns={table_to_xbrl_factoid_name_dict[name]: "xbrl_factoid"}
+        )
+        for (name, df) in kwargs.items()
+    }
+
+    other_dimensions = get_other_dimensions()
+    dimensions = (
+        pd.concat(tbls.values())[["table_name", "xbrl_factoid"] + other_dimensions]
+        .drop_duplicates()
+        .reset_index(drop=True)
+    )
+    return dimensions
+
+
+@asset(
+    ins={
+        "clean_xbrl_metadata_json": AssetIn("clean_xbrl_metadata_json"),
+        "table_dimensions_ferc1": AssetIn("table_dimensions_ferc1"),
+    }
+)
+def calculation_components_xbrl_ferc1(**kwargs):
+    """Create calculation-compnent table from table-level metadata."""
+    clean_xbrl_metadata_json = kwargs["clean_xbrl_metadata_json"]
+    table_dimensions = kwargs["table_dimensions_ferc1"]
+    # compile all of the calc comp tables.
+    calc_metas = []
+    for table_name, tranformer in FERC1_TFR_CLASSES.items():
+        calc_meta = (
+            tranformer(xbrl_metadata_json=clean_xbrl_metadata_json[table_name])
+            .process_xbrl_metadata_calculations()
+            .assign(table_name=table_name)
+        )
+        calc_metas.append(calc_meta)
+    # squish all of the calc comp tables then add in the implicit table dimensions
+    calc_components = pd.concat(calc_metas).pipe(
+        make_implict_dimensions_explict_in_calculaiton_components, table_dimensions
+    )
+    return calc_components
+
+
+def make_implict_dimensions_explict_in_calculaiton_components(
+    calculation_components: pd.DataFrame, table_dimensions: pd.DataFrame
+) -> pd.DataFrame:
+    """Make all of the implied dimensions in the calculation components explict.
+
+    We are assuming that if a dimension was associted with a calculation component, then
+    that is the only dimension that applies whereas if no dimension was associated with
+    a calculation component, that calculation component should be applied across all
+    relevant dimensions.
+
+    Relate all of the calculation components with all found dimensions from
+    :func:`table_dimensions_ferc1`, unless the calculation components already
+    contained information regarding dimensions.
```

Comment:
I'm not understanding what this function is doing from the docstring or the code.

What do we expect the structure of the input calculation component df to look like, and how do we change it? What does it mean to "relate all of the calculation components with all found dimensions?"

Reply:
okay i added a lot of backstory to the docs.. does that make it more clear?